database/migrations/pgsql/initial.sql

```diff
@@ -0,0 +1,632 @@
+ALTER TABLE public.audit_logs ADD CONSTRAINT audit_logs_user_id_foreign FOREIGN KEY (user_id) REFERENCES public.users(id) ON DELETE SET NULL ON UPDATE RESTRICT;
+
+
+-- public.backups foreign keys
+
+ALTER TABLE public.backups ADD CONSTRAINT backups_server_id_foreign FOREIGN KEY (server_id) REFERENCES public.servers(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+
+
+-- public.databases foreign keys
+
+ALTER TABLE public.databases ADD CONSTRAINT databases_database_host_id_foreign FOREIGN KEY (database_host_id) REFERENCES public.database_hosts(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+ALTER TABLE public.databases ADD CONSTRAINT databases_server_id_foreign FOREIGN KEY (server_id) REFERENCES public.servers(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+
+
+-- public.schedules foreign keys
+
+ALTER TABLE public.schedules ADD CONSTRAINT schedules_server_id_foreign FOREIGN KEY (server_id) REFERENCES public.servers(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+
+
+-- public.server_transfers foreign keys
+
+ALTER TABLE public.server_transfers ADD CONSTRAINT server_transfers_server_id_foreign FOREIGN KEY (server_id) REFERENCES public.servers(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+
+
+-- public.server_variables foreign keys
+
+ALTER TABLE public.server_variables ADD CONSTRAINT server_variables_server_id_foreign FOREIGN KEY (server_id) REFERENCES public.servers(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+ALTER TABLE public.server_variables ADD CONSTRAINT server_variables_variable_id_foreign FOREIGN KEY (variable_id) REFERENCES public.egg_variables(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+
+
+-- public.servers foreign keys
+
+ALTER TABLE public.servers ADD CONSTRAINT servers_allocation_id_foreign FOREIGN KEY (allocation_id) REFERENCES public.allocations(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+ALTER TABLE public.servers ADD CONSTRAINT servers_egg_id_foreign FOREIGN KEY (egg_id) REFERENCES public.eggs(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+ALTER TABLE public.servers ADD CONSTRAINT servers_nest_id_foreign FOREIGN KEY (nest_id) REFERENCES public.nests(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+ALTER TABLE public.servers ADD CONSTRAINT servers_node_id_foreign FOREIGN KEY (node_id) REFERENCES public.nodes(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+ALTER TABLE public.servers ADD CONSTRAINT servers_owner_id_foreign FOREIGN KEY (owner_id) REFERENCES public.users(id) ON DELETE RESTRICT ON UPDATE RESTRICT;
+
+
+-- public.subusers foreign keys
+
+ALTER TABLE public.subusers ADD CONSTRAINT subusers_server_id_foreign FOREIGN KEY (server_id) REFERENCES public.servers(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+ALTER TABLE public.subusers ADD CONSTRAINT subusers_user_id_foreign FOREIGN KEY (user_id) REFERENCES public.users(id) ON DELETE CASCADE ON UPDATE RESTRICT;
+
+
+-- public.tasks foreign keys
+
+ALTER TABLE public.tasks ADD CONSTRAINT tasks_schedule_id_foreign FOREIGN KEY (schedule_id) REFERENCES public.schedules(id) ON DELETE CASCADE ON UPDATE RESTRICT;
```

Comment:
As Laravel supports PostgreSQL, out-of-the-box this shouldn't be necessary, as Eloquent and the query builder, knows how to form it. The migrations should work just fine with Laravel & PostgreSQL.